src/DotVVM.Framework/Resources/Scripts/DotVVM.ts

```diff
@@ -659,39 +670,37 @@ class DotVVM {
         });
     }
 
-    public handleSpaNavigation(element: HTMLElement): boolean {
+    public handleSpaNavigation(element: HTMLElement) {
         var target = element.getAttribute('target');
-
         if (target == "_blank") {
             return true;
```

Comment:
Here you return `boolean`, later you return a `Promise`. I guess that's not what you intended.